lib/report_generator.rb

```diff
@@ -9,13 +9,13 @@ def self.generate(query_id, query_params, recipients, opts = {})
       recipients = filter_recipients_by_query_access(recipients, query)
       params = params_to_hash(query_params)
 
-      result = DataExplorer.run_query(query, params)[:pg_result]
+      result = DataExplorer.run_query(query, params)
```

Comment:
I'm thinking if we can promote the return value from a hash to a class or struct instead. Then we could implement for example `result.empty?` and use that on L:15. It could also implement `result.convert` which could be delegated to the new converter class. WDYT?

Reply:
Maybe? But I'm not sure. This result structure should be used in many places. Changing it will affect many files.